src/clj_puppetdb/http.clj

```diff
@@ -82,12 +82,15 @@
   of `java.lang.Throwable`) and rethrowing them as :exception in the `exception-structure`."
   [call exception-structure & exceptions]
   (let [exceptions (if (empty? exceptions) [Throwable] exceptions)]
-    `(try
-       ~call
-       ~@(map (fn [exception]
-                `(catch ~exception exception#
-                   (throw (ex-info nil (assoc ~exception-structure :exception exception#)))))
-              exceptions))))
+    `(let [response# (try
+                      ~call
+                      ~@(map (fn [exception]
+                               `(catch ~exception exception#
+                                  (throw (ex-info nil (assoc ~exception-structure :exception exception#)))))
+                             exceptions))]
+      (if-let [exception# (:error response#)]
+        (throw (ex-info nil (assoc ~exception-structure :exception exception#)))
```

Comment:
Since this is the same as line 89, I think it could be moved into a `let` as a partial or anonymous function. My guideline for this kind of thing is usually if something is used three times it should definitely go into a `let` binding, if it's used twice it could go into a `let` but it's not absolutely necessary. I leave it up to you here as to what you think is best. :)